src/axl.c

```diff
@@ -573,9 +637,8 @@ int AXL_Dispatch (int id)
         AXL_ERR("Could not find transfer info for UID %d", id);
         return AXL_FAILURE;
     }
-
     /* check that handle is in correct state to dispatch */
-    if (xstate != AXL_XFER_STATE_CREATED) {
+    if (xstate != AXL_XFER_STATE_CREATED && !resume) {
```

Comment:
We dropped the check on the state when resuming.  I know resume needs to deal with both CREATED and DISPATCHED, but this might open things up for the user to call in with other states like COMPLETED.  Do the methods for all of the various backends work for all possible states?

Reply:
Good point. What do you think about these per-state actions if you call AXL_Resume():

AXL_XFER_STATE_NULL: return error
AXL_XFER_STATE_WAITING: Try to resume where you left off.
AXL_XFER_STATE_COMPLETED: Don't do anything.  Return success.
AXL_XFER_STATE_CANCELED: Try to resume where you left off.